spec/rspec/mocks/matchers/receive_spec.rb

```diff
@@ -90,6 +90,16 @@ module Mocks
           }.to raise_error(UnsupportedMatcherError)
         end
 
+        it 'does support inherited matchers', :unless => options.include?(:allow_other_matchers) do
+          receive_foo = Class.new(RSpec::Mocks::Matchers::Receive).new(:foo, nil)
+          object = double
+
+          allow(object).to receive_foo
+          expect(object).to receive_foo
+
+          object.foo
+        end
```

Comment:
This is a shared spec that is meant to run in 4 different contexts:

* `expect(object).to receive_foo`
* `allow(object).to receive_foo`
* `expect_any_instance_of(object).to receive_foo`
* `allow_any_instance_of(object).to receive_foo`

But as you've written it, it tests just the first 2 of these, and does so for all 4 contexts this shared spec is run in.  Instead, I think it should be:

``` ruby

        it 'supports inherited matchers' do
          receive_foo = Class.new(RSpec::Mocks::Matchers::Receive).new(:foo, nil)
          wrapped.to receive_foo
          receiver.foo
        end
```

`wrapped` is polymorphic and will return `allow(object)`, `expect(object)` etc.  `receiver` will return `object` or an instance of the class passed to `allow_any_instance_of(klass)`/`expect_any_instance_of(klass)` -- so by using these, you can write one spec and have it operate for all the contexts appropriately.